dolphinscheduler-ui/src/views/security/user-manage/components/use-authorize.ts

```diff
@@ -265,10 +324,42 @@ export function useAuthorize() {
         })
       })
 
+      let fullPathFileIdWithReadPerm = []
+      const pathFileIdWithReadPerm: Array<string> = []
+      state.authorizedFileResourcesWithReadPerm.forEach((v: number) => {
+        state.fileResources.forEach((v1: any) => {
+          const arr = []
+          arr[0] = v1
+          if (getParent(arr, v).length > 0) {
+            fullPathFileIdWithReadPerm = getParent(arr, v).map((v2: any) => {
+              return v2.id
+            })
+            pathFileIdWithReadPerm.push(fullPathFileIdWithReadPerm.join('-'))
+          }
+        })
+      })
+
+      let fullPathUdfIdWithReadPerm = []
+      const pathUdfIdWithReadPerm: Array<string> = []
+      state.authorizedUdfResourcesWithReadPerm.forEach((v: number) => {
+        state.udfResources.forEach((v1: any) => {
+          const arr = []
+          arr[0] = v1
+          if (getParent(arr, v).length > 0) {
+            fullPathUdfIdWithReadPerm = getParent(arr, v).map((v2: any) => {
+              return v2.id
+            })
+            pathUdfIdWithReadPerm.push(fullPathUdfIdWithReadPerm.join('-'))
+          }
+        })
+      })
```

Comment:
This block is similar to the previous one. It's better to do this by a function.